trusts/decorators.py

```diff
@@ -119,4 +194,31 @@ def check_perms(request, *args, **kwargs):
             raise PermissionDenied
         return False
 
+    def check_by_p(request, *args, **kwargs):
+        perms = perm.get_leaves()
+
+        for p in perms:
+            fieldlookups = _resolve_fieldlookups(request, kwargs, **p.get_lookup_kwargs())
+            items = None
+            if fieldlookups is not None:
+                items = _get_permissible_items(request, p.perm, fieldlookups)
+                if items is None:
+                    p.permitted = False
```

Comment:
Not sure it is a good idea to modify P instance. P is being reused for each http request.